.github/workflows/ci.yml

```diff
@@ -61,8 +61,19 @@ jobs:
         with:
           tool: tomlq
 
+      - name: Enforce libp2p-swarm-derive version is pinned and matches latest version
+        if: env.CRATE == 'libp2p-swarm-derive'
+        run: |
+          PACKAGE_VERSION=$(cargo metadata --format-version=1 --no-deps | jq -e -r '.packages[] | select(.name == "'"$CRATE"'") | .version')
+          SPECIFIED_VERSION=$(tomlq "workspace.dependencies.$CRATE.version" --file ./Cargo.toml)
+
+          echo "Package version: $PACKAGE_VERSION";
+          echo "Specified version: $SPECIFIED_VERSION";
+
+          test "=$PACKAGE_VERSION" = "$SPECIFIED_VERSION"
```

Comment:
I think instead adding another check, we could also just use `||`:

```shell
test "$PACKAGE_VERSION" = "$SPECIFIED_VERSION" || test "=$PACKAGE_VERSION" = "$SPECIFIED_VERSION"
```

What do you think?

Reply:
That is a bit more general than we specifically need but semantically the same. All this check should enforce is that we keep the versions in sync. If we ever end up pinning another crate, it should likely apply there too!

Reply:
sure, I'll fix this.

Reply:
nvrmnd, you did it already.